src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -522,15 +522,10 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
 
             if (result == null) {
                 // non-protocol related error
-                String errMessage = null;
-                byte[] message = response.getBody();
-                if (message != null) {
-                    errMessage = new String(message);
-                } else {
-                    errMessage = "Status code:" + String.valueOf(response.getStatusCode());
-                }
+                final String errMessage = response.getBody() == null ? 
```

Comment:
It would be useful to check of empty size in here. I.e. `response.getBody() == null || response.getBody().size() == 0`